test/nvexec/then.cpp

```diff
@@ -201,12 +202,95 @@ TEST_CASE("nvexec then can return values of non-trivial types", "[cuda][stream][
   REQUIRE(flags_storage.all_set_once());
 }
 
-TEST_CASE("nvexec then can preceed a sender with values", "[cuda][stream][adaptors][then]") {
+class tracer_storage_t {
+  int h_counter_storage{};
+  int *h_counter_{};
+  int *d_counter_{};
+
+public:
+  tracer_storage_t() : h_counter_{&h_counter_storage} {
+    cudaMalloc(&d_counter_, sizeof(int));
+    cudaMemset(d_counter_, 0, sizeof(int));
+  }
+
+  ~tracer_storage_t() {
+    cudaFree(d_counter_);
+  }
+
+  class handle_t {
+    int *h_counter_{};
+    int *d_counter_{};
+
+    handle_t(int *h_counter, int *d_counter)
+      : h_counter_{h_counter}
+      , d_counter_{d_counter} {
+    }
+
+    void diff(int val) {
+      cuda::std::atomic_ref<int> ref{*(is_on_gpu() ? d_counter_ : h_counter_)};
+      ref.fetch_add(val, cuda::std::memory_order_relaxed);
+    }
+
+  public:
+    void more() {
+      diff(+1);
+    }
+
+    void less() {
```

Comment:
```suggestion
    void fewer() {
```

Sorry, my inner grammar nazi got the best of me. ;-)